go/libkb/chain_link_v2.go

```diff
@@ -63,7 +63,33 @@ func (t SigchainV2Type) NeedsSignature() bool {
 	case SigchainV2TypeTrack, SigchainV2TypeUntrack, SigchainV2TypeAnnouncement:
 		return false
 	default:
+		// Unsupported types don't need signatures. Otherwise we can't
+		// make code forwards-compatible.
+		return t.IsSupportedType()
+	}
+}
+
+func (t SigchainV2Type) IsSupportedType() bool {
```

Comment:
This function kind of seems like a trap. Would two separate `IsSupportedUserType` and `IsSupportedTeamType`  work?

Reply:
ok sure